### PR TITLE
[camera_windows] Check string size before Win32 MultiByte <-> WideChar conversions

### PR DESCRIPTION
**Pull request by tgucio**
This PR adds a check for string max length in Utf8FromUtf16 and Utf16FromUtf8, similar to https://github.com/flutter/flutter/pull/99729 / https://github.com/flutter/flutter/pull/94608.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/camera/camera_windows/pubspec.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ name: camera_windows
 description: A Flutter plugin for getting information about and controlling the camera on Windows.
 repository: https://github.com/flutter/plugins/tree/main/packages/camera/camera_windows
 issue_tracker: https://github.com/flutter/flutter/issues?q=is%3Aissue+is%3Aopen+label%3A%22p%3A+camera%22
-version: 0.2.0
+version: 0.2.0+1
```
Comment (cbracken) [inline packages/camera/camera_windows/pubspec.yaml:5]: This should be 0.2.1.

Ref: https://dart.dev/tools/pub/versioning#semantic-versions

> For simplicity’s sake, avoid using + after the version reaches 1.0.0.

**Review by cbracken — CHANGES_REQUESTED** [review]
Did you encounter a bug that led to this patch? If so, please list it in the PR description and add a test that replicates the failure.

File: packages/camera/camera_windows/pubspec.yaml (hunk 1 above)
Comment (tgucio) [reply packages/camera/camera_windows/pubspec.yaml:5]: ACK - changing to 0.2.1.

**Comment by tgucio** [comment]
@cbracken not really a particular issue - same story as https://github.com/flutter/flutter/pull/94608 (found through code audit vs. Win32 API docs).

**Comment by cbracken** [comment]
Thanks. Normally we'd ask for a test, but I'm not sure we necessarily want to add a test that involves passing a string longer than `std::string::max_size` which on my machine is 9 exabytes, or roughly 1 million American Libraries of Congress. While this does update the code to implement safe behaviour according to the API docs, I'd imagine that in practice, this isn't actually an issue on the latest 64-bit releases of Windows. It will need a test exemption though.

**Comment by Hixie** [comment]
test-exempt: not practically testable

**Review by cbracken — APPROVED** [review]
Thanks for your contribution.

LGTM stamp from a Japanese personal seal

**Comment by tgucio** [comment]
@cbracken Great, camera_windows will be safe against 1+ million American Libraries of Congress string attacks :-).
Could you slap the automerge label as I don't have write access?

I see the submit-queue/CI failures in other plugin PRs so I'll sync from main to see if it unclogs things.

**Comment by cbracken** [comment]
Thanks!

**Pull request merged**